schematic/manifest/generator.py

```diff
@@ -1606,12 +1631,17 @@ def _update_dataframe_with_existing_df(self, empty_manifest_url: str, existing_d
                 **dict(zip(new_columns, len(new_columns) * [""]))
             )
 
+        end_columns = self._get_end_columns(current_schema_headers=current_schema_headers,
+                                           existing_manifest_headers=existing_manifest_headers,
+                                           out_of_schema_columns=out_of_schema_columns)
+        
         # sort columns in the updated manifest:
         # match latest schema order
         # move obsolete columns at the end
         updated_df = updated_df[self.sort_manifest_fields(updated_df.columns)]
-        updated_df = updated_df[[c for c in updated_df if c not in out_of_schema_columns] + list(out_of_schema_columns)]
 
+        # move obsolete columns at the end with entityId at the very end
```

Comment:
hmm I feel like there are currently three places of updating the order of manifests: 
* call `_get_end_columns` function
*  updated_df = updated_df[self.sort_manifest_fields(updated_df.columns)]
* `updated_df = updated_df[[c for c in updated_df if c not in end_columns] + list(end_columns)]`
Is there a way to consolidate them so that the code become simplier?